dotcom-rendering/src/web/components/SeriesSectionLink.tsx

```diff
@@ -170,6 +170,29 @@ const immersiveTitleBadgeStyle = (palette: Palette) => css`
 		6px 0 0 0 ${palette.background.seriesTitle};
 `;
 
+const pickSeriesTag = (tags: TagType[]): TagType | undefined => {
```

Comment:
Is there a need to abstract this? Could the logic just live in the `SeriesSectionLink` components? e.g.
```
const observerTag = tags.find(
  (tag) => tag.type === 'Publication' && tag.title === 'The Observer',
);
const isCommentIsFree = tags.some((tag) => tag.id === 'commentisfree/commentisfree')
const seriesTag = tags.find(
  (tag) =>
    tag.type === 'Blog' ||
    tag.type === 'Series' ||
    (tag.type === 'Publication' &&
      tag.title === 'The Observer'),
);

// Observer opinion (commentisfree) articles should prioritise
// the publication tag over the commentisfree tag.
const tag = observerTag && isCommentIsFree ? observerTag : seriesTag;

```

Reply:
This is much tidier, thank you :)